### PR TITLE
HHH-12604 Replace EqualsHelper.equals with Objects.equals

### PR DESCRIPTION
**Pull request by marschall**
https://hibernate.atlassian.net/browse/HHH-12604

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java [hunk 1]
```diff
@@ -16,6 +16,16 @@
...
+	 * @param x the first object, possibly null
+	 * @param y the second object, possibly null
+	 * @return true if the two objects are equal either by identity
+	 *         or using their equals method
+	 * @deprecated use {@link java.util.Objects#equals(Object, Object)}
+	 */
+	@Deprecated
 	@SuppressWarnings("SimplifiableIfStatement")
 	public static boolean equals(final Object x, final Object y) {
 		if ( x == y ) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java]: IMO this should be changed to call `Objects#equals`

Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java]: @sebersole considering it's internal and shouldn't be used anymore in our code, shouldn't we simply remove it?

**Comment by sebersole** [comment]
One or the other.  But just deprecating it misses the point imo

File: hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java (hunk 1 above)
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java]: @marschall could you simply remove this method from the code? The idea is to not use it anymore in our code and people shouldn't use our internel helpers either.

Thanks!

Comment (marschall) [reply hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java]: Sure, no problem.

**Comment by gsmet** [comment]
Rebased and merged. Thanks @marschall .

FYI, we don't usually add a link to the JIRA in the commit message (we just have the reference at the beginning of the commit message). I amended the commit to remove it.

**Pull request closed without merging**